Hosted onboarding (#536)
mollie/api-documentation: source/onboarding/getting-started.rst

There's also the `status`: ``needs-data``, ``in-review`` and ``completed``

As a basic implementation, we recommend handling the following cases:

1) Payments are **not** enabled and the status is ``needs-data``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: false,
        canReceiveSettlements: false,
        status: "needs-data"
      }

  .. list-table::
    :widths: auto

    * - | ``Key information``
      - `Before you can accept live payments, Mollie needs more information`

    * - | ``Action``
      - `Add more information` (link to Onboarding URL)

2) Payments are enabled and the status is ``needs-data``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: true,
        canReceiveSettlements: false,
        status: "needs-data"
      }

  .. list-table::
    :widths: auto

    * - | ``Key information``
      - `You are ready to start accepting your first payments. For complete functionality complete your account.` (Here you can tell them about receiving settlements.)

    * - | ``Action``

>       - `Add more information` (link to Onboarding URL)

Reviewer: ```suggestion
      - `Complete your Mollie account` (link to Onboarding URL)
```

3) Payments are **not** enabled and the status is ``in-review``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: false,
        canReceiveSettlements: false,
        status: "in-review"
      }

  .. list-table::
    :widths: auto

    * - | ``Key information``

>       - `You’ve supplied us with all information we need. Mollie is verifying your details.`

Reviewer: ```suggestion
      - `Mollie has all the required information and is verifying your details.`
```

The :doc:`Onboarding status </reference/v2/onboarding-api/get-onboarding-status>` response is used to display a proper message and action to your customer.

There's two milestones that the customer reaches during the Mollie onboarding, ``canReceivePayments`` (basic information has been provided) and ``canReceiveSettlements`` (all information has been provided and verified).

There's also the `status`: ``needs-data``, ``in-review`` and ``completed``

As a basic implementation, we recommend handling the following cases:

1) Payments are **not** enabled and the status is ``needs-data``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: false,
        canReceiveSettlements: false,
        status: "needs-data"
      }

  .. list-table::
    :widths: auto

    * - | ``Key information``
      - `Before you can accept payments, Mollie needs more information.`

    * - | ``Action``
      - `Continue with Mollie` (link to Onboarding URL)

2) Payments are enabled and the status is ``needs-data``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: true,
        canReceiveSettlements: false,
        status: "needs-data"
      }

>   .. note::  If you are only interested in your merchants accepting payments, you should redirect them to your default landing page.

Reviewer: ```suggestion
  .. note::  If you are only interested in your merchants accepting payments, you should redirect them to your default landing page at this time.
```

Step 2: Customer signs up and gives authorization
-------------------------------------------------
Once you send your customer to the authorize URL they will see a welcome screen.
This page informs the customer that the application is using Mollie to create payments and displays the logo you've
configured in the `hosted onboarding settings <https://www.mollie.com/dashboard/settings/hosted-onboarding>`_.

When the customer continues they will be shown a signup form (they can also login if they already have an account). Then, the oAuth permission screen will ask for permissions to view their onboarding status and submit data.

.. image:: ../oauth/images/oauth-permission-onboarding@2x.png

Step 3: Customer returns from authorization
-------------------------------------------
After the authorization, the customer will be sent to the redirect URL provided in the `hosted onboarding settings
<https://www.mollie.com/dashboard/settings/hosted-onboarding>`_.

  .. note::  With successful authorization a query string will be added to the redirect URL, for example: ``https://www.yourapp.com/payments/settings?code=bvS9VpCVbvBrQVfSdG9F3aNtWszdQnzz``

> You can now generate the oAuth access tokens for this customer using the :doc:`Generate tokens endpoint </reference/oauth2/tokens>` and store them. You will use this token for all the requests regarding that customer.

Author: We refer to the docs already about, here it'd be a bit overwhelming imo.

Step 4: Customer starts onboarding
-------------------------------------------

You can now check the Onboarding Status of your
customer using the :doc:`Get onboarding status endpoint </reference/v2/onboarding-api/get-onboarding-status>`

We recommend to handle the status of the customer like shown in the following flowchart:

.. image:: ./images/merchant-status.png

If your customer just authorized with Mollie, you probably want the customer to immediately start the Onboarding process with Mollie.

> In that case, the customer will be returning from the oauth flow and the response status from the previous request will be ``needs-data``.

Author: But it's indeed the first time they authorize with this app.

  .. note::  To make your customer experience easier, you can help them onboard at Mollie by prefilling some fields using the :doc:`Submit onboarding data endpoint </reference/v2/onboarding-api/submit-onboarding-data>`

Step 5: Customer needs to complete the onboarding
----------------------------------------------------------------
The :doc:`Onboarding status </reference/v2/onboarding-api/get-onboarding-status>` response is used to display a proper message and action to your customer.

There's two milestones that the customer reaches during the Mollie onboarding, ``canReceivePayments`` (basic information has been provided) and ``canReceiveSettlements`` (all information has been provided and verified).

There's also the `status`: ``needs-data``, ``in-review`` and ``completed``

As a basic implementation, we recommend handling the following cases:

1) Payments are NOT enabled and the status is ``needs-data``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: false,
        canReceiveSettlements: false,

>         status: needs-data

Reviewer: This will come back as string right?

      }

  .. list-table::
    :widths: auto

    * - | ``Key information``
      - `Before you can accept live payments, Mollie needs more information`

    * - | ``Action``
      - `Add more information` (link to Onboarding URL)

2) Payments are enabled and the status is ``needs-data``

  .. code-block:: javascript
    :linenos:

      {
        canReceivePayments: true,
        canReceiveSettlements: false,
        status: needs-data
      }

  .. list-table::
    :widths: auto

    * - | ``Key information``

>       - `You are ready to start accepting your first payments. To receive payments in your bank account complete your account.`

Reviewer: It really depends on the partner, a lot of them will not care about settlements.

    * - | ``Action``
      - `Add more information` (link to Onboarding URL)

> 3) Payments are NOT enabled and the status is ``in-review``

Reviewer: ```suggestion
3) Payments are **not** enabled and the status is ``in-review``
```
